Tests/AssemblyToProcessTests.cs

```diff
@@ -285,20 +336,39 @@ public void OnChangedMethodAttributeCustomizesCalledMethods()
         var instance = testResult.GetInstance(nameof(ClassWithOnChangedCustomized));
         instance.Property1 = "foo";
 
+        DumpWarnings(nameof(ClassWithOnChangedCustomized));
+
         Assert.False(instance.OnProperty1ChangedCalled);
         Assert.True(instance.FirstCustomCalled);
         Assert.True(instance.SecondCustomCalled);
-        Assert.DoesNotContain(testResult.Warnings, w => w.Text.ContainsWholeWord(nameof(ClassWithOnChangedCustomized)));
```

Comment:
I wouldn't remove these asserts (applies to all places with `//Warnings tested in ClassWithOnChangedCustomizedWarnings`)

Reply:
The asserts are no longer correct, in that there is now one warning expected.  The real requirement for this now is that there is exactly one warning produced, and it's about OnProperty1Changed not being called.  So I could copy/paste this 3 times into the methods commented with `//Warnings tested in ClassWithOnChangedCustomizedWarnings)` but thought it best to make this an independent test.  I've changed it a little now so will commit the following soon:
```    
[Fact]
    public void ClassWithOnChangedCustomizedWarnings()
    {
        var warnings = testResult.Warnings
            .Where(w => w.Text.ContainsWholeWord(nameof(ClassWithOnChangedCustomized)))
            .ToArray();

        Assert.Contains(warnings, w => w.Text.ContainsWholeWord(nameof(ClassWithOnChangedCustomized))
                                       && w.Text.Contains(nameof(ClassWithOnChangedCustomized.OnProperty1Changed)));

        Assert.True(warnings.Length == 1);
    }


```

Reply:
Just as further justification here, I found when fixing the bug that I generated other incorrect warnings before getting the correct solution.  A single assertion that I *do* have a certain warning message doesn't mean I don't have a bug generating 10 other completely incorrect warning messages.  Hence the need to verify just a single warning, with the correct content.